libs/libcommon/include/common/DateLUTImpl.h

```diff
@@ -118,15 +121,9 @@ class DateLUTImpl
         return lut[DayNum(index - (lut[index].day_of_week - 1))].date;
     }
 
-    inline DayNum toFirstDayNumOfWeek(DayNum d) const
-    {
-        return DayNum(d - (lut[d].day_of_week - 1));
-    }
+    inline DayNum toFirstDayNumOfWeek(DayNum d) const { return DayNum(d - (lut[d].day_of_week - 1)); }
```

Comment:
For consistency the functions `toFirstDayNumOfWeek()` and `toStartOfWeekInterval()` should get the new parameter `week_mode` too to find out whether weeks starts on Sunday or on Monday.

Reply:
Let me do it.

Reply:
@vitlibar 
The default value of week_mode is 0, which is the first day of the week on Sunday.
`toFirstDayNumOfWeek()` is called `toMonday` for users, and if you add new parameter week_mode, it will output Sunday or Monday, which is confusing.

[https://clickhouse.yandex/docs/en/query_language/functions/date_time_functions/#tomonday](https://clickhouse.yandex/docs/en/query_language/functions/date_time_functions/#tomonday)
``` c++
struct ToMondayImpl
{
    static constexpr auto name = "toMonday";

    static inline UInt16 execute(UInt32 t, const DateLUTImpl & time_zone)
    {
        return time_zone.toFirstDayNumOfWeek(time_zone.toDayNum(t));
    }
```

Does default week_mode value need to be added to the system settings? 
The default value is recommended to be 3, so that we can remain the consistency of the original implementation.

toStartOfWeekInterval() itself has several parameters, isn't it too complicated for users to add week_mode?  There is no week interval calculation in the document.
[https://clickhouse.yandex/docs/en/query_language/functions/date_time_functions/#tostartofinterval-time-or-data-interval-x-unit-time-zone](https://clickhouse.yandex/docs/en/query_language/functions/date_time_functions/#tostartofinterval-time-or-data-interval-x-unit-time-zone)
`toStartOfInterval(time_or_data, INTERVAL x unit [, time_zone])`

**Summary:**
1. add new function：`toStartOfWeek(t[,mode])` to find out whether week start on Sunday or on Monday.
2. Add a `week_mode `configuration item to the system settings, default value of week_mode is 3.
3. `toStartOfInterval `does not change any more. 

Can you see if this is ok?

Reply:
Ok, looks good. Why do you think the default value for `default_week_mode` should be `3`? In MySQL it's `0`.